src/main/java/com/openshift/jenkins/plugins/pipeline/model/IOpenShiftImageTagger.java

```diff
@@ -17,272 +17,274 @@
-
-						String token = destToken;
-						if ( token == null || token.isEmpty() ) {
-							token = getAuthToken();
-						}
-
-						OpenShiftCreator isCreator = new OpenShiftCreator(getApiURL(newOverrides), destNS, token, ""+chatty, createJson);
-						isCreator.setAuth(Auth.createInstance(chatty ? listener : null, getApiURL(newOverrides), overrides));
-
-						boolean newISCreated = isCreator.coreLogic(launcher, listener, newOverrides);
-
-						if (!newISCreated) {
-							listener.getLogger().println(String.format(MessageConstants.EXIT_TAG_CANNOT_CREATE_DEST_IS, destStream, destNS));
-							return false;
-						}
-
-						destIS = srcClient.get(ResourceKind.IMAGE_STREAM, destStream, destNS);
-					}
-
-					if (destIS == null) {
-						listener.getLogger().println(String.format(MessageConstants.EXIT_TAG_CANNOT_GET_IS, destStream, destNS));
-						return false;
-					}
-
-					destIS.addTag( destTag, tagType, srcImageID, srcNS);
-					if (chatty) {
-						listener.getLogger().println("\n updated image stream json " + ((ImageStream) destIS).getNode().toJSONString(false) + " with tag: " + destTag );
-					}
-
-					try {
-						destClient.update(destIS);
-						break;
-					} catch ( OpenShiftException ose ) {
-						if ( ose.getStatus().getCode() == 403 && retries > 1 ) { // If resource was updating, retry
-							continue;
-						}
-						throw ose;
-					}
-				}
-			}
-
-	    	listener.getLogger().println(String.format(MessageConstants.EXIT_OK, DISPLAY_NAME));
-			return true;
-    	} else {
-    		return false;
-    	}
-
-	}
+    final static String DISPLAY_NAME = "Tag OpenShift Image";
+
+    default String getDisplayName() {
+        return DISPLAY_NAME;
+    }
+
+    String getAlias();
+
+    String getSrcTag();
+
+    String getDestTag();
+
+    String getSrcStream();
+
+    String getDestStream();
+
+    String getDestinationNamespace();
+
+    String getDestinationAuthToken();
+
+    default String getAlias(Map<String, String> overrides) {
+        return getOverride(getAlias(), overrides);
+    }
+
+    default String getSrcTag(Map<String, String> overrides) {
+        String val = getOverride(getSrcTag(), overrides);
+        // If there is no source namespace, default to jenkins namespace
+        if (val.length() == 0 && overrides != null && overrides.containsKey(NAMESPACE_ENV_VAR)) {
+            val = overrides.get(NAMESPACE_ENV_VAR);
+        }
+        return val;
+
+    }
+
+    default String getDestTag(Map<String, String> overrides) {
+        return getOverride(getDestTag(), overrides);
+    }
+
+    default String getSrcStream(Map<String, String> overrides) {
+        return getOverride(getSrcStream(), overrides);
+    }
+
+    default String getDestStream(Map<String, String> overrides) {
+        return getOverride(getDestStream(), overrides);
+    }
+
+    default String getDestinationNamespace(Map<String, String> overrides) {
+        return getOverride(getDestinationNamespace(), overrides);
```

Comment:
correct .... per the discussion @jupierce and I had on IRC, this needs to be
1) call getOverride(getDestin....), if non-empty return
2) return getNamespace(overrides)
getNamespace(overrides) should handle the fetching from NAMESPACE_ENV_VAR